src/prefect/server/models/artifacts.py

```diff
@@ -172,6 +184,44 @@ async def delete_artifact(
     Returns:
         bool: True if the delete was successful, False otherwise
     """
+    current_artifact = await session.get(db.Artifact, artifact_id)
+    if current_artifact is None:
+        return False
+
+    is_latest_artifact_version = (
+        await session.execute(
+            sa.select(db.ArtifactCollection)
+            .where(db.ArtifactCollection.key == current_artifact.key)
+            .where(db.ArtifactCollection.latest_id == artifact_id)
+        )
+    ).scalar_one_or_none()
+
+    if is_latest_artifact_version is not None:
```

Comment:
If this is `is_latest_...` maybe we should do the `is not None` check above so this is a bool? Then this if statement reads clearer.

Reply:
I found the logic here a little challenging to follow, I think mostly because of the variable names so I have a few suggestions here. A comment describing that we update the latest artifact to point to the next latest version if it exists would be helpful too.

Reply:
Yes, that sounds clearer 👍